### PR TITLE
tunelinux: retrieve chapinfo

### PR DESCRIPTION
**Pull request by raunakkumar**
- retrieve chap settings from chap.json config file
- iscsi target update node to include authmethod

Signed-off-by: Raunak Kumar <rkumar@nimblestorage.com>

### CODE REVIEW

File: tunelinux/iscsi.go [hunk 1]
```diff
@@ -46,7 +48,7 @@ var (
 		"noop_out_timeout":    noopOutTimeoutPattern,
 		"noop_out_interval":   noopOutTimeoutIntervalPattern,
 		"replacement_timeout": replacementTimeoutPattern,
-		"nr_sessions":         nrSessionsPattern}
```
Comment (shivamerla) [inline tunelinux/iscsi.go:49]: this is not used anywhere?

File: linux/iscsi.go [hunk 2]
```diff
@@ -45,6 +46,7 @@ const (
 	chapPasswordMaxLen      = 16
 	nodeChapUser            = "node.session.auth.username"
 	nodeChapPassword        = "node.session.auth.password"
+	nodeChapAuthmethod      = "node.session.auth.authmethod"
```
Comment (rgcostea) [inline linux/iscsi.go:49]: nit: Wouldn't nodeChapAuthMethod be correct?

File: linux/iscsi.go [hunk 3]
```diff
@@ -329,6 +337,23 @@ func updateConnectionMode(target *model.IscsiTarget, targets model.IscsiTargets,
 	return
 }
 
+// update chap authmethod
+func updateChapAuthmethod(target *model.IscsiTarget) (err error) {
```
Comment (rgcostea) [inline linux/iscsi.go:341]: same here: updateChapAuthMethod

**Comment by rgcostea** [comment]
Adding @sneharai4  as a reviewer.

**Comment by raunakkumar** [comment]
Closing this PR in preference to configMap restricted within CSI Driver instead of tunelinux

**Pull request closed without merging**